### PR TITLE
Fix some warnings

### PR DESCRIPTION
**Pull request by remilauzier**
Most of it is to fix hidden anonymous lifetime that are deprecated and will be remove in rust 2021.

### CODE REVIEW

File: crates/vergen/src/lib.rs [hunk 1]
```diff
@@ -83,6 +83,12 @@
 //! See the [generate_version_rs](crate::output::codegen::generate_version_rs) documentation
 //! if you wish to use this method.
 #![deny(
+  bare_trait_objects,
+  rust_2018_compatibility,
+  rust_2018_idioms,
+  nonstandard_style,
+  future_incompatible,
+  unused,
```
Comment (tdaede) [inline crates/vergen/src/lib.rs:91]: Isn't unused default?

**Comment by lu-zero** [comment]
It does not seem to survive the CI, I'd not merge this until later.

File: crates/vergen/src/lib.rs (hunk 1 above)
Comment (remilauzier) [reply crates/vergen/src/lib.rs:91]: Not all the lints are deny by default.

**Comment by remilauzier** [comment]
On my end, i can't see why the Ci hit the timeout. Do you see the error that did that on your end?

**Comment by lu-zero** [comment]
It unhelpfully states
```
Run timed out after > 72 hours.
```

I'd postpone this patch till the compiler is more stable and the edition is set in stone, if it isn't something you already built upon other patches.

**Comment by remilauzier** [comment]
Can you tried to restart the CI pass on your end, to see if the result will be different?

**Comment by remilauzier** [comment]
I have made the CI pass.

**Pull request closed without merging**